### PR TITLE
Michael ayele prefixing prrte binaries

### PR DESCRIPTION
**Pull request by Michael-uofl**
Updated Makefile.am files to include @PRTE_BINARY_PREFIX@ in library
and binary names

### CODE REVIEW

File: src/tools/prted/Makefile.am
Comment (jsquyres) [inline src/tools/prted/Makefile.am:51]: Missed adding the prefix to the 2 lines above this -- I'll fix.

File: config/prte_configure_options.m4
Comment (jsquyres) [inline config/prte_configure_options.m4]: This statement is incorrect (i.e., it's incomplete).  I'll fix.

**Review by jsquyres — DISMISSED** [review]

File: src/Makefile.am
Comment (jsquyres) [inline src/Makefile.am]: Missed updating these, too.  I'll fix.

**Comment by jsquyres** [comment]
* Restored the `prte` command
* Fixed some places where `PRTE_BINARY_PREFIX` was missed
* Fixed a mistake in the `.m4` file
* Took the upstream fix from https://github.com/openpmix/prrte/pull/2087

@hppritcha Please review

**Comment by hppritcha** [comment]
trying to build this branch and autogen.pl is failing
```
libtoolize: copying file 'config/ltsugar.m4'
libtoolize: copying file 'config/ltversion.m4'
libtoolize: copying file 'config/lt~obsolete.m4'
autoreconf: running: /usr/bin/autoconf --include=config --include=config/oac --force --warnings=all,no-obsolete,no-override
autoreconf: running: /usr/bin/autoheader --include=config --include=config/oac --force --warnings=all,no-obsolete,no-override
autoheader: cannot rename /tmp/aramcX1V/ahjFCanl/config.hin as 
autoheader: 
autoheader:     include/prte_version.h
autoheader: .in: No such file or directory
autoreconf: /usr/bin/autoheader failed with exit status: 1
Command failed: autoreconf -ivf --warnings=all,no-obsolete,no-override -I config -I config/oac
```

**Comment by hppritcha** [comment]
pushing a small fix to this branch shortly.  Please squash it before merging.

**Review by hppritcha — APPROVED** [review]
I approve assuming the SQUASHME commit is included.

**Comment by jsquyres** [comment]
@hppritcha There was a standalone commit that made that `configure.ac` change -- I just dropped it (and your SQUASHME).  Should be good now.

**Pull request merged**
